connect/runtime/src/test/java/org/apache/kafka/connect/integration/ErrorHandlingIntegrationTest.java

```diff
@@ -172,6 +173,17 @@ public void testSkipRetryAndDLQWithHeaders() throws Exception {
         connect.deleteConnector(CONNECTOR_NAME);
     }
 
+    private boolean partitionsAssigned() {
+        try {
+            ConnectorStateInfo info = connect.connectorStatus(CONNECTOR_NAME);
+            return info != null && info.tasks().size() == NUM_TASKS
+                    && connectorHandle.taskHandle(TASK_ID).partitionsAssigned() == 1;
+        }  catch (Exception e) {
+            log.error("Could not check connector state info. Swallowing exception.", e);
+            return false;
```

Comment:
The "Swallowing the exception" is a bit alarming at first. It's true that if this method returns false then `TestUtils.waitForCondition` (which is using this method as the test condition) will fail at line 137.

Pretty minor, but how about the following?
```
            // Log the exception and return that the partitions were not assigned
            log.error("Could not check connector state info.", e);
            return false;
```